src/v/redpanda/admin/api-doc/usage.json

```diff
@@ -24,8 +23,34 @@
         }
     ],
     "models": {
-        "usage_response": {
-            "id": "usage_response",
+        "follower_usage_response": {
```

Comment:
I found this pretty confusing: to fill out a whole different object depending on whether we're controller leader or not.  It's almost all the same data.

Relying on controller leadership for this is a bit fragile in general: that leadership will move around, so in principle a client can end up getting two leader-style responses from different nodes.  This would be simpler if the client would just query different URLs for node-local vs cluster-wide data

Reply:
just to give you an overview of our thinking, we're all open to hear better ideas how to express this.

we currently have 3 metrics but they mean different things - ingress/egress is per node, so consumer of that data has to sum it up to get the full value. The third one is storage, which is reported only from single node and for whole cluster.

So the first problem is that there's bunch of metrics but they mean different things (we could get around this by proper naming though).

the other thing was that we did not want followers to return `storage: 0` for storage just because they don't know the value. So first proposal was to make storage nullable and return null from followers. But that is not an option for swagger I understoood. So we ended up proposing this

```
global {
storage: 1,
},
per-broker {
egress: 1,
ingress: 1
}
```

where global would be served only by leader who really have those aggregated values.

Better ideas welcome :)